students/DiannaTingg/lessons/lesson07/assignment/findings.txt

```diff
@@ -0,0 +1,22 @@
+"""
+Lesson 07 Assignment
+Notes
+"""
+
+I refactored the code from Lesson 05, so now the program returns a list of two tuples, one for customers and one for products.
+Each tuple contains: number of records processed, record count in the database prior to running, record count after running, and time taken to run the module.
+
+I'm using larger csv files for products, customers, and rentals.  Each file has 9,999 records plus a header row.
+I automatically clear the MongoDB each time the program exits.
+
+Using timeit, my linear.py file takes 2.573601484 seconds to run once.  It takes 25.389773754 seconds to run 10 times.
+I wrote some tests in test_linear.py and they all pass.
+
+I'm going to use multithreading in my parallel.py file to see if I can get a performance improvement.
+I will read all three csv files at the same time (concurrently).
+
+Using timeit, my parallel.py file takes 1.679744485 seconds to run once.  It takes 18.042013637 seconds to run 10 times.
+I wrote some tests in test_parallel.py and they all pass.
+
+I keep getting different times from timeit, but in general, parallel.py seems to be faster than linear.py, so I would recommend using that program instead.
+The program never seems to fail due to contention.
```

Comment:
Whether or not there is much speed increase depends on how mongodb is implemented. Since all parallel calls are going to mongodb, if it then takes them more or less linearly, it won't make  a huge difference. The fact that it improved the time a bit, means it is a bit better that. Probably if we were using a network database with lots of workers, we would see more improvement.